### PR TITLE
Fix launching Azure Eventhubs emulator on linux

### LINKED ISSUE #5326: `Azure.EventHubs` emulator crashes on startup on Linux due to permission issues
This can be seen when running `playground/AspireEventHub/EventHubs.AppHost`, and the dashboard shows the `eventhubns` having crashed with:

```
Unhandled exception. System.UnauthorizedAccessException: Access to the path '/Eventhubs_Emulator/ConfigFiles/Config.json' is denied.
 ---> System.IO.IOException: Permission denied
   --- End of inner exception stack trace ---
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirError)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode, Boolean failForSymlink, Boolean& wasSymlink, Func`4 createOpenException)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, UnixFileMode openPermissions, Int64& fileLength, UnixFileMode& filePermissions, Boolean failForSymlink, Boolean& wasSymlink, Func`4 createOpenException)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at Microsoft.Extensions.Configuration.FileConfigurationProvider.<Load>g__OpenRead|6_0(IFileInfo fileInfo)
   at Microsoft.Extensions.Configuration.FileConfigurationProvider.Load(Boolean reload)
   at Microsoft.Extensions.Configuration.ConfigurationRoot..ctor(IList`1 providers)
   at Microsoft.Extensions.Configuration.ConfigurationBuilder.Build()
   at Microsoft.Extensions.Hosting.HostBuilder.InitializeAppConfiguration()
   at Microsoft.Extensions.Hosting.HostBuilder.Build()
   at a.a.aaR.A(String[])
   at a.a.aaR.<Main>(String[])
```

I was able to reproduce this in a test in `Aspire.Hosting.Azure.Tests` with:
```csharp
    {
        using var builder = TestDistributedApplicationBuilder.Create().WithTestAndResourceLogging(testOutput);
        var blob = builder.AddAzureStorage("ehstorage")
            .RunAsEmulator()
            .AddBlobs("checkpoints");
        var eventHubs = builder.AddAzureEventHubs("eh").RunAsEmulator().AddEventHub("hub");
        using var app = builder.Build();

        CancellationTokenSource cts = new();
        var t = app.RunAsync();

        var hb = Host.CreateApplicationBuilder();
        hb.AddAzureEventHubConsumerClient("eventhubns", settings =>
        {
            settings.EventHubName = "hub";
        });
        using var host = hb.Build();
        var t2 = host.RunAsync();
        
        await Task.Delay(500_000);
        await app.StopAsync();
    }
```

Here the file is created by aspire:
https://github.com/dotnet/aspire/blob/c31eaf3f814014884df7ee70db3a6d5cfbd75d0c/src/Aspire.Hosting.Azure.EventHubs/AzureEventHubsExtensions.cs#L145-L150
https://github.com/dotnet/aspire/blob/c31eaf3f814014884df7ee70db3a6d5cfbd75d0c/src/Aspire.Hosting.Azure.EventHubs/AzureEventHubsExtensions.cs#L187-L191

cc @eerhardt @sebastienros another one of the file permissions in container when running from a Linux host.

### PR DESCRIPTION
**Pull request by radical**
Starting the emulator on linux fails with:

```
Unhandled exception. System.UnauthorizedAccessException: Access to the path '/Eventhubs_Emulator/ConfigFiles/Config.json' is denied.
 ---> System.IO.IOException: Permission denied
   --- End of inner exception stack trace ---
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirError)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode, Boolean failForSymlink, Boolean& wasSymlink, Func`4 createOpenException)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, UnixFileMode openPermissions, Int64& fileLength, UnixFileMode& filePermissions, Boolean failForSymlink, Boolean& wasSymlink, Func`4 createOpenException)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at Microsoft.Extensions.Configuration.FileConfigurationProvider.<Load>g__OpenRead|6_0(IFileInfo fileInfo)
   at Microsoft.Extensions.Configuration.FileConfigurationProvider.Load(Boolean reload)
   at Microsoft.Extensions.Configuration.ConfigurationRoot..ctor(IList`1 providers)
   at Microsoft.Extensions.Configuration.ConfigurationBuilder.Build()
   at Microsoft.Extensions.Hosting.HostBuilder.InitializeAppConfiguration()
   at Microsoft.Extensions.Hosting.HostBuilder.Build()
   at a.a.aaR.A(String[])
   at a.a.aaR.<Main>(String[])
```

Fixes https://github.com/dotnet/aspire/issues/5326 .

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/5358)

### CODE REVIEW

**Comment by radical** [comment]
I'm unable to add a proper test for this because on CI, with the fix it fails with:
```
fail: a.a.aaW[0]
		Emulator Start up probe Unsuccessful. MetadataStore Health status: Unhealthy BlobStore Health status: Unhealthy
```

.. which is a different issue to fix!

File: src/Aspire.Hosting.Azure.EventHubs/AzureEventHubsExtensions.cs [hunk 1]
```diff
@@ -133,6 +134,15 @@ public static IResourceBuilder<AzureEventHubsResource> RunAsEmulator(this IResou
         }
 
         // Add emulator container
+        var configHostFile = Path.GetTempFileName();
+        File.WriteAllBytes(configHostFile, []);
+        if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```
Comment (eerhardt) [inline src/Aspire.Hosting.Azure.EventHubs/AzureEventHubsExtensions.cs:139]:
```suggestion
        if (!OperatingSystem.IsWindows())
```

File: src/Aspire.Hosting.Azure.EventHubs/AzureEventHubsExtensions.cs [hunk 2]
```diff
@@ -133,6 +134,15 @@ public static IResourceBuilder<AzureEventHubsResource> RunAsEmulator(this IResou
         }
 
         // Add emulator container
+        var configHostFile = Path.GetTempFileName();
+        File.WriteAllBytes(configHostFile, []);
```
Comment (eerhardt) [inline src/Aspire.Hosting.Azure.EventHubs/AzureEventHubsExtensions.cs:138]: Why is this necessary? `Path.GetTempFileName` is documented as

https://learn.microsoft.com/en-us/dotnet/api/system.io.path.gettempfilename

> Creates a uniquely named, zero-byte temporary file on disk and returns the full path of that file.

**Review by eerhardt — APPROVED** [review]
LGTM. It would be good to get a test here eventually.
